feature/gribi/otg_tests/gribi_scaling/README.md

```diff
@@ -10,34 +10,30 @@ Validate gRIBI scaling requirements.
 *   Create 64 L3 sub-interfaces under DUT port-2 and corresponding 64 L3
     sub-interfaces on ATE port-2
 *   On DUT port-1 and ATE port-1 create a single L3 interface
-*   On DUT, create a policy-based forwarding rule to redirect all traffic received from DUT port-1 into VRF-1 (based on src. IP match criteria)
+*   On DUT, create a policy-based forwarding rule to redirect all traffic received from DUT port-1 into VRF-1 (based on src. IP match 111.111.111.111)
+*   On DUT, create a policy-based forwarding rule to redirect all traffic received from DUT port-1 into VRF-2 (based on src. IP match 222.222.222.222)
 *   Establish gRIBI client connection with DUT negotiating FIBACK as the
     requested ack_type and make it become leader.
-*   Using gRIBI Modify RPC install the following IPv4Entry sets, and validate
+*   TODO: Using gRIBI Modify RPC install the following IPv4Entry sets, and validate
     the specified behaviours:
-    *   <Default VRF> IPv4Entries -> NHG -> Multiple NH.
-        *   Inject IPv4Entries(IPBlockDefaultVRF: 198.18.196.1/22) in default
-            VRF
-        *   Install 64 L3 sub-interfaces IP to NextHopGroup containing one
-            NextHop specified to ATE port-2.
-        *   Validate that the entries are installed as FIB_PROGRAMMED
-    *   <VRF1> IPv4Entries -> Multiple NHG -> Multiple NH.
-        *   Inject IPv4Entries(IPBlock1: "198.18.0.1/18") in VRF1.
-        *   Install 1000 IPs from IPBlockDefaultVRF to 10 NextHopGroups
-            containing 100 NextHops each
-        *   Validate that the entries are installed as FIB_PROGRAMMED
-    *   <VRF2> IPv4Entries -> Multiple NHG -> Multiple NH.
-        *   Inject IPv4Entries(IPBlock2: "198.18.64.1/18") in VRF2.
-        *   Install *repeat* 17.5K NH from 1K /32 from IPBlockDefaultVRF to 35
-            NextHopGroups containing 45 NextHops each
-        *   Validate that the entries are installed as FIB_PROGRAMMED
-    *   <VRF3> IPv4Entries -> Multiple NHG -> Multiple NH.
-        *   Inject IPv4Entries(IPBlock3: "198.18.128.1/18") in VRF3.
-        *   Install IPiniP decap-then-encap to 500 first /32 from <IPBlockVRF1>
-            to 500 NextHopGroups containing 1 NextHop each
-        *   Validate that the entries are installed as FIB_PROGRAMMED
+    *   <Default VRF>
+        * A) Install 400 NextHops, egressing out different interfaces.
+        * B) Install 200 NextHopGroups.  Each points at 8 NextHops from the first 200 entries of A) with equal weight.
+        * C) Install 200 IPv4 Entries, each pointing at a unique NHG (1:1) from B.
+        * D) Install 200 NextHops.  Each will redirect to an IP from C).
+        * E) Install 100 NextHopGroups.  Each will contain 2 NextHops from D) with weights 1:31. The backup next_hop_group will be to redirect to VRF3.
```

Comment:
How are we going to validate the weight distribution  ? finally all packets goes out of same physical interface . May be it is good to  have different egress interface for  second half of B) NHGs and use that for second half of C) prefixes . Build E with one NH from first half D) and 1NH from second half of D).

Reply:
Yes, that seems reasonable.  Updated.